### DISCUSSION TITLE
segment size will change to zero after calling ReleaseCollection

### DISCUSSION THREAD

**Discussion by xiaojuan8** [milvus-io/milvus#16040] [Q&A and General discussion]
Hi, I build an index, and it shows 8 segments when building index finished(I called GetQuerySegmentInfo to get segment size).  later, I released this collection from query node memory. Then, when I load this collection again, it shows 0 segment.(I called GetQuerySegmentInfo to get segment size).  So I am confused wether it is a bug, or I missed some details? 

Round shows 8 segments, Round 2 shows 0 segments. Both loading collection returns OK.

My cpp code :
-----------------------------------------
   Round 1:

    const auto progress_monitor = ::milvus::ProgressMonitor::NoWait();
    status = client->LoadCollection(collection_name, progress_monitor);
    LOG(INFO) << "load collection status : " << status.IsOk();

    ::milvus::QuerySegmentsInfo info;
    status = client->GetQuerySegmentInfo(collection_name, info);
    std::cout << "segment size : " << info.size() << std::endl; 
     
     //xxxxxx
     
     status = client->ReleaseCollection(collection_name);
------------------------------------------
     Round 2:
     
    const auto progress_monitor = ::milvus::ProgressMonitor::NoWait();
    status = client->LoadCollection(collection_name, progress_monitor);
    LOG(INFO) << "load collection status : " << status.IsOk();

    ::milvus::QuerySegmentsInfo info;
    status = client->GetQuerySegmentInfo(collection_name, info);
    std::cout << "segment size : " << info.size() << std::endl;

**Comment by yhmo** [comment]
If you input ::milvus::ProgressMonitor::NoWait() for LoadCollection(), it will instantly return without checking the loading progress.
You can try the ::milvus::ProgressMonitor::Forever()
```
const auto progress_monitor = ::milvus::ProgressMonitor::Forever();
status = client->LoadCollection(collection_name, progress_monitor);
```

**Comment by yhmo** [comment]
If you input ::milvus::ProgressMonitor::NoWait() for LoadCollection(), it will instantly return without checking the loading progress.
You can try the ::milvus::ProgressMonitor::Forever()
```
const auto progress_monitor = ::milvus::ProgressMonitor::Forever();
status = client->LoadCollection(collection_name, progress_monitor);
```

**Reply by xiaojuan8** [reply]
👌， Thanks for reply! Nice.
